src/time.c

```diff
@@ -39,13 +39,68 @@ rcutils_time_point_value_as_nanoseconds_string(
   if (0 == str_size) {
     return RCUTILS_RET_OK;
   }
-  if (rcutils_snprintf(str, str_size, "%.19" PRId64, *time_point) < 0) {
+  if (rcutils_snprintf(str, str_size, "%.19" PRId64, *time_point) <= 0) {
```

Comment:
why this change is needed? i think negative value will be returned in any error state. it depends on `vsnprintf` which possibly returns zero? besides, this change is not related to this PR enhancement?

Reply:
Whoops, that was a search and replace problem, will fix it together with the below comment